app/src/main/java/com/yacgroup/yacguide/utils/SearchBarHandler.kt

```diff
@@ -22,38 +23,45 @@ import android.content.SharedPreferences
 import android.text.Editable
 import android.text.TextWatcher
 import android.view.View
+import android.view.View.VISIBLE
 import android.view.inputmethod.InputMethodManager
 import android.widget.CheckBox
 import android.widget.EditText
 import androidx.constraintlayout.widget.ConstraintLayout
+import com.google.android.material.textfield.TextInputLayout
 import com.yacgroup.yacguide.R
 
 class SearchBarHandler(searchBarLayout: ConstraintLayout,
                        searchHintResource: Int,
-                       checkBoxTitle: String,
-                       checkBoxDefaultValue: Boolean,
-                       private val _settings: SharedPreferences,
-                       private val _initCallback: (Boolean) -> Unit,
-                       private val _updateCallback: (String, Boolean) -> Unit) {
+                       checkBoxTitle: String = "",
+                       checkBoxDefaultValue: Boolean = false,
+                       private val _settings: SharedPreferences? = null,
+                       private val _initCallback: ((Boolean) -> Unit)? = null,
+                       private val _updateCallback: ((String, Boolean) -> Unit)? = null,
+                       checkBoxVisibility: Int = VISIBLE) {
```

Comment:
- [ ] Please use `View.INVISIBLE` and remove the `android.view.View.INVISIBLE` import
- [ ] Please put this parameter 3 lines above (behind checkBoxDefaultValue)

Reply:
Because the class is used without named arguments in the code, I wanted to make it compatible. But anyway, I will change those lines, too.